rotkehlchen/chain/ethereum/adex/typing.py

```diff
@@ -0,0 +1,310 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Callable, Dict, List, NamedTuple, Optional, Tuple, Union
+
+from eth_typing.evm import ChecksumAddress
+from eth_utils.typing import HexStr
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.fval import FVal
+from rotkehlchen.typing import Timestamp
+
+AdexEventDBTuple = (
+    Tuple[
+        str,  # tx_hash
+        str,  # address
+        str,  # identity_address
+        int,  # timestamp
+        str,  # type
+        str,  # pool_id
+        str,  # amount
+        str,  # usd_value
+        Optional[str],  # bond_id
+        Optional[int],  # nonce
+        Optional[int],  # slashed_at
+        Optional[int],  # unlock_at
+        Optional[str],  # channel_id
+    ]
+)
+
+
+class EventType(Enum):
+    """Supported events"""
+    BOND = 1
+    UNBOND = 2
+    UNBOND_REQUEST = 3
+    CHANNEL_WITHDRAW = 4
+
+    def __str__(self) -> str:
+        if self == EventType.BOND:
+            return 'bond'
+        if self == EventType.UNBOND:
+            return 'unbond'
+        if self == EventType.UNBOND_REQUEST:
+            return 'unbond_request'
+        if self == EventType.CHANNEL_WITHDRAW:
+            return 'channel_withdraw'
+        raise AttributeError(f'Corrupt value {self} for EventType -- Should never happen')
+
+    def pretty_name(self) -> str:
+        if self == EventType.BOND:
+            return 'deposit'
+        if self == EventType.UNBOND:
+            return 'withdraw'
+        if self == EventType.UNBOND_REQUEST:
+            return 'withdraw request'
+        if self == EventType.CHANNEL_WITHDRAW:
+            return 'claim'
+        raise AttributeError(f'Corrupt value {self} for EventType -- Should never happen')
+
+
+@dataclass(init=True, repr=True)
+class Bond:
+    tx_hash: HexStr  # from bond.id
+    address: ChecksumAddress
+    identity_address: ChecksumAddress
+    timestamp: Timestamp
+    bond_id: HexStr
+    pool_id: HexStr
+    value: Balance
+    nonce: int
+    slashed_at: Timestamp  # from bond.slashedAtStart
+
+    def serialize(self) -> Dict[str, Any]:
+        from .utils import POOL_ID_POOL_NAME
```

Comment:
please never import code from inside a function like this. Especially multiple times like this in a file.

I see you tried to do this to go around cyclic imports. But this is not solving the problem, just complicating it. You should just sanitize code so you don't have cyclic imports.

Reply:
Fixed it. Please resolve when you read it.